be/src/olap/tablet.cpp

```diff
@@ -2988,11 +2997,12 @@ Status Tablet::calc_segment_delete_bitmap(RowsetSharedPtr rowset,
             }
 
             // sequence id smaller than the previous one, so delete current row
-            if (st.is<KEY_ALREADY_EXISTS>()) {
+            if (st.is<KEY_ALREADY_EXISTS>() && (!is_partial_update || have_input_seq_column)) {
```

Comment:
Please also update the comment here, it's a little hard to understand for other devs